From 85c005ee1345c342f707f3c55317daf6cb050603 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org.ua>
Date: Thu, 10 Jan 2019 18:18:49 +0200
Subject: Remove erroneous abort() call

The call was introduced by commit ccef8581. It caused tar to abort
on perfectly normal operations, like untarring archives containing
./ with the -U option,

See http://lists.gnu.org/archive/html/bug-tar/2019-01/msg00019.html
for details.

* src/extract.c (maybe_recoverable): Remove misplaced call to abort().
---
 src/extract.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/extract.c b/src/extract.c
index 090b866..8276f8f 100644
--- a/src/extract.c
+++ b/src/extract.c
@@ -787,7 +787,7 @@ maybe_recoverable (char *file_name, bool regular, bool *interdir_made)
 	case UNLINK_FIRST_OLD_FILES:
 	  break;
 	}
-      abort (); /* notreached */
+      FALLTHROUGH;
 
     case ENOENT:
       /* Attempt creating missing intermediate directories.  */
-- 
cgit v1.0-41-gc330

